var/spack/repos/builtin/packages/picsar/package.py

```diff
@@ -0,0 +1,133 @@
+##############################################################################
+# Copyright (c) 2018, Los Alamos National Security, LLC.
+# Produced at the Los Alamos National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+class Picsar(MakefilePackage):
+    """
+    The Particle-In-Cell Scalable Application Resource (PICSAR) is a high performance repository intended to help scientists porting their 
+    Particle-In-Cell (PIC) codes to the next generation of exascale computers.
+
+    PICSAR exploits the three levels of parallelism that will be required to achieve good performances on future architectures: distributed memory 
+    parallelization (internode), shared memory parallelization (intranode) and vectorization.
+
+    PICSAR includes:
+
+    A high performance library of highly optimized versions of the key functionalities of the PIC loop.
+    A compact "mini-app" standalone code, which serves as a self-contained proxy that adequately portrays the computational loads and dataflow of 
+    more complex PIC codes.
+    A Python wrapper for using PICSAR optimized routines with Python-driven codes.
+    """
+
+    homepage = "http://picsar.net"
+    git      = "https://bitbucket.org/berkeleylab/picsar.git"
+
+    version('develop', branch='PICSARlite')
+   
+    variant('gnu', default=True, description='Build with gnu compiler')
```

Comment:
Compilers are part of the Spack "spec". Don't add them as variant.

  https://spack.readthedocs.io/en/latest/basic_usage.html#compiler-specifier

If PICSAR only supports specific flags for those compilers, you can add them below implicitly, e.g. `if '%gcc' in self.spec`